src/test/groovy/com/onesignal/androidsdk/MainTest.groovy

```diff
@@ -825,6 +825,71 @@ class MainTest extends Specification {
         }
     }
 
+    def 'fcm 15 and crashlytics with firebase platform BOM'() {
+        when:
+        def results = runGradleProject([
+            'android.useAndroidX': true,
+            skipGradleVersion: GRADLE_OLDEST_VERSION,
+            compileLines : """\
+                implementation platform('com.google.firebase:firebase-bom:25.4.1')
+                implementation 'com.google.firebase:firebase-messaging:15.0.0'
+                implementation 'com.google.firebase:firebase-crashlytics'
+            """
+        ])
+
+        then:
+        assert results // Asserting existence and contains 1+ entries
+        results.each {
+            // 1. Ensure firebase-bom is allowed to upgrade firebase-messaging
+            assert it.value.contains('com.google.firebase:firebase-messaging:15.0.0 -> 20.2.0')
+            // 2. Ensure version from firebase-bom is used and we don't try to downgrade
+            assert it.value.contains('com.google.firebase:firebase-crashlytics -> 17.0.1')
+        }
+    }
+
+    def 'pre-15 firebase with firebase-bom platform 17 BOM'() {
+        when:
+        def results = runGradleProject([
+            'android.useAndroidX': true,
```

Comment:
Does `'android.enableJetifier': true,` need to be added as well since Android X is `true`?

Reply:
Good question but no, this test does not have any libraries in it also using the Android Support library.